src/crate/client/http.py

```diff
@@ -87,6 +88,12 @@ def default(self, o):
 class Server(object):
 
     def __init__(self, server, **kwargs):
+        self.auth = None
+        if "@" in server:
+            auth = server.split("@")
+            if "//" in auth[0]:
+                auth = auth[0].split("//")[1]
+            self.auth = auth.split(":")
```

Comment:
wouldn't it make sense to already prepare the complete header entry here? Otherwise some part of the work is done here and the rest is done on each request.